src/metabase/lib/native.cljc

```diff
@@ -249,3 +249,11 @@
   "Returns the native query's template tags"
   [query :- ::lib.schema/query]
   (:template-tags (lib.util/query-stage query 0)))
+
+(mu/defn has-write-permission :- :boolean
+  "Returns whether the database has native write permissions.
+   This is only filled in by [[metabase.api.database/add-native-perms-info]]
+   and added to metadata when pulling a database from the list of dbs in js."
```

Comment:
Can this be formulated more explicitly? It's not 100% clear from the text what role `metabase.api.database/add-native-perms-info` is playing. This is describing the preconditions for this function to make sense, right?